node/peer.js

```diff
@@ -45,11 +45,22 @@ function TChannelPeer(channel, hostPort, options) {
     self.state = null; // TODO
     self.connections = [];
     self.random = self.channel.random;
+
+    self.stateOptions = {
+        timers: self.channel.timers,
+        random: self.channel.random,
+        period: self.options.period,
+        maxErrorRate: self.options.maxErrorRate,
+        probation: self.options.probation,
+        stateMachine: self,
```

Comment:
Why call the peer the state machine and the next handler ?

Why not just call it the peer.